library/Fission/AWS.hs

```diff
@@ -0,0 +1,52 @@
+module Fission.AWS
+  ( createEnv
+  , ensureContent
+  , validate
+  , withAWS
+  ) where
+
+import RIO
+
+import Data.Has
+import Servant
+
+import Network.AWS
+import Network.AWS.Auth as AWS
+import Network.AWS.Route53
+
+import qualified Fission.Config as Config
+import           Fission.Internal.Constraint
+import           Fission.Web.Error
+
+withAWS :: (MonadUnliftIO m, HasEnv r) => r -> Region -> AWS a -> m a
+withAWS env region = runResourceT . runAWS env . within region
+
+createEnv :: MonadRIO           cfg m
+          => Has AWS.AccessKey cfg
+          => Has AWS.SecretKey cfg
+          => m Env
+createEnv = do
+  accessKey :: AccessKey <- Config.get
+  secretKey :: SecretKey <- Config.get
+  liftIO $ newEnv $ FromKeys accessKey secretKey
+
+validate :: ChangeResourceRecordSetsResponse -> Either ServerError ChangeResourceRecordSetsResponse
+validate changeSet =
+  if status >= 300
+    then Left $ toServerError status
```

Comment:
🤔 300s are redirects and caching. I guess it can't hurt to make that an error, because the HTTP client should have hit that first and I would not expect a 300 to get this far into the app.